python/test/unit/language/test_core.py

```diff
@@ -169,43 +169,63 @@ class MmaLayout:
     def __init__(self, version, warps_per_cta, ctas_per_cga, cta_split_num, cta_order, instr_shape):
         self.version = version
         self.warps_per_cta = warps_per_cta
-        self.ctas_per_cga = str(ctas_per_cga)
-        self.cta_split_num = str(cta_split_num)
-        self.cta_order = str(cta_order)
-        self.instr_shape = str(instr_shape)
+        self.ctas_per_cga = ctas_per_cga
+        self.cta_split_num = cta_split_num
+        self.cta_order = cta_order
+        self.instr_shape = instr_shape
 
     def __str__(self):
-        return f"#{GPU_DIALECT}.nvidia_mma<{{versionMajor={self.version[0]}, versionMinor={self.version[1]}, warpsPerCTA={str(self.warps_per_cta)}, CTAsPerCGA={self.ctas_per_cga}, CTASplitNum={self.cta_split_num}, CTAOrder={self.cta_order}, instrShape={self.instr_shape}}}>"
+        version = self.version
+        warps_per_cta = str(self.warps_per_cta)
+        ctas_per_cga = str(self.ctas_per_cga)
+        cta_split_num = str(self.cta_split_num)
+        cta_order = str(self.cta_order)
+        instr_shape = str(self.instr_shape)
+        return f"#{GPU_DIALECT}.nvidia_mma<{{versionMajor={version[0]}, versionMinor={version[1]}, warpsPerCTA={warps_per_cta}, CTAsPerCGA={ctas_per_cga}, CTASplitNum={cta_split_num}, CTAOrder={cta_order}, instrShape={instr_shape}}}>"
```

Comment:
you can append `!s` to apply `str()` inside `{}`, e.g. `f'{self.warps_per_cta!s}'`

Reply:
Good catch, thank you!

Turned out that no conversions is needed at all, so I removed this part. I think now change looks better